flow/workflows/cdc_flow.go

```diff
@@ -493,37 +535,48 @@ func CDCFlowWorkflowWithConfig(
 				" limit on the number of syncflows to be executed: ", currentSyncFlowNum)
 			break
 		}
-		currentSyncFlowNum++
+		currentSyncFlowNum += 1
+		w.logger.Info("executing sync flow", slog.Int("count", currentSyncFlowNum), slog.String("flowName", cfg.FlowJobName))
 
-		syncFlowID, err := GetChildWorkflowID(ctx, "sync-flow", cfg.FlowJobName)
-		if err != nil {
-			finishNormalize()
-			return state, err
+		syncMetaCtx := workflow.WithLocalActivityOptions(ctx, workflow.LocalActivityOptions{
+			StartToCloseTimeout: 1 * time.Minute,
+		})
+		lastSyncInput := &protos.GetLastSyncedIDInput{
+			PeerConnectionConfig: cfg.Destination,
+			FlowJobName:          cfg.FlowJobName,
 		}
 
-		// execute the sync flow as a child workflow
-		childSyncFlowOpts := workflow.ChildWorkflowOptions{
-			WorkflowID:        syncFlowID,
-			ParentClosePolicy: enums.PARENT_CLOSE_POLICY_REQUEST_CANCEL,
-			RetryPolicy: &temporal.RetryPolicy{
-				MaximumAttempts: 20,
-			},
-			SearchAttributes:    mirrorNameSearch,
-			WaitForCancellation: true,
+		lastSyncFuture := workflow.ExecuteLocalActivity(syncMetaCtx, flowable.GetLastSyncedID, lastSyncInput)
```

Comment:
@serprex  should we move `GetLastSyncedId` into `StartFlow`, I think this moves us closer to the world where we flush to destination more frequently and retries pickup from a newer point.

Reply:
Done, it's cleaner, but not sure it helps in that regard. Better would be to have getoffset/setoffset managed by source connector